### PR TITLE
domain/rg: organize and enhance the code of RunawayChecker

### PR DESCRIPTION
**Pull request by JmPotato**
### What problem does this PR solve?

Issue Number: ref #54434.

### What changed and how does it work?

- Delete some unnecessary code.
- Add more comments.
- Organize and enhance some functions.

### Check List

Tests 

- [x] Unit test
- [x] Integration test

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/domain/resourcegroup/runaway.go [hunk 1]
```diff
@@ -499,50 +483,48 @@ func (r *RunawayChecker) BeforeExecutor() error {
...
-			r.markRunaway(RunawayMatchTypeWatch, action, &now)
-			// If no match action, it will do nothing.
-			switch action {
-			case rmpb.RunawayAction_Kill:
-				return exeerrors.ErrResourceGroupQueryRunawayQuarantine
-			case rmpb.RunawayAction_CoolDown:
-				// This action should be done in BeforeCopRequest.
-				return nil
-			case rmpb.RunawayAction_DryRun:
-				return nil
```
Comment (HuSharp) [inline pkg/domain/resourcegroup/runaway.go:520]: why removed `RunawayAction_DryRun`?

File: pkg/domain/resourcegroup/runaway.go [hunk 2]
```diff
@@ -596,25 +569,43 @@ func (r *RunawayChecker) CheckCopRespError(err error) error {
...
+	r.watchAction = action
+	now := time.Now()
+	r.markRunaway("watch", action, &now)
 }
 
-func (r *RunawayChecker) markRunaway(matchType RunawayMatchType, action rmpb.RunawayAction, now *time.Time) {
-	actionStr := strings.ToLower(rmpb.RunawayAction_name[int32(action)])
-	metrics.RunawayCheckerCounter.WithLabelValues(r.resourceGroupName, matchType.String(), actionStr).Inc()
+func (r *RunawayChecker) markRunaway(matchType string, action rmpb.RunawayAction, now *time.Time) {
+	actionStr := strings.ToLower(action.String())
```
Comment (HuSharp) [inline pkg/domain/resourcegroup/runaway.go:599]: please change this line as well :)
https://github.com/pingcap/tidb/blob/7291829726b1c112e1179469c679b3bcd684eabd/pkg/executor/infoschema_reader.go#L3626

File: pkg/domain/resourcegroup/runaway.go (hunk 1 above)
Comment (JmPotato) [reply pkg/domain/resourcegroup/runaway.go:520]: It can be covered by the `default` case, there is no need to explicitly write it here I think

Comment (HuSharp) [reply pkg/domain/resourcegroup/runaway.go:520]: I agree that this action can be similar to the default case.
> Previous behavior may have been wrong:
> Previously, the dryRun branch would return directly, but now it iterates over all identifiers in a for loop.

There is a concept of [dryRun](https://docs.pingcap.com/tidb/stable/tidb-resource-control#query_limit-parameters) in the documentation, but after removing this `RunawayAction_DryRun` in the code,  there is no dryRun to be found in the code (which may be confusing?). Do we need to put it here explicitly? Just like: 
```
		case rmpb.RunawayAction_DryRun:
		default:
```

Comment (JmPotato) [reply pkg/domain/resourcegroup/runaway.go:520]: Good catch, you're right!

**Comment by JmPotato** [comment]
/cc @nolouch @glorv @HuSharp 

PTAL

**Review by HuSharp — APPROVED** [review]

**Review by glorv — APPROVED** [review]

**Comment by D3Hunter** [comment]
/approve

for domain pkg

**Review by nolouch — APPROVED** [review]

**Pull request merged**